### DISCUSSION TITLE
How to know 1st sync is done?

### DISCUSSION THREAD

**Discussion by sylverbee** [Mimetis/Dotmim.Sync#447] [Q&A]
How to get syncStage=EndSession?

**Comment by Mimetis** [comment]
`EndSession` is raised just before end of sync.

You can catch if from your `LocalOrchestrator` for example:

``` cs
agent.LocalOrchestrator.OnSessionEnd(sessionEndArgs => {
});
```
You can even catch it from the `IProgress<ProgressArgs>` :

``` cs
progress = new SynchronousProgress<ProgressArgs>(s =>
{
    Console.WriteLine($"{s.PogressPercentageString}:\t{s.Source}:\t{s.Message}");

    if (s.Context.SyncStage == SyncStage.EndSession)
    {
        Console.WriteLine("Reach End of Session");
    }
});
    
```

Is it what you need ?

**Reply by sylverbee** [reply]
My need is to wait for the end of 1st before moving forward

**Reply by Mimetis** [reply]
IS not enough ?

**Reply by sylverbee** [reply]
Nop, but I found another solution.
